db/handle_buf.c

```diff
@@ -138,6 +138,39 @@ static void thd_io_complete(void)
     UNLOCK(&lock);
 }
 
+
+#define LISTC_CLEAN(listp, lnk, dofree, type) { \
+    type *item, *tmp; \
+    /* free each item */ \
+    LISTC_FOR_EACH_SAFE(listp, item, tmp, lnk) { \
+        /* remove and potentially free item */ \
+        listc_rfl(listp, item); \
+        if (dofree) \
+            free(item); \
+    } \
+}
+
+
+void thd_cleanup()
+{
+    int counter = 0;
+    while (nthdcreates - nretire && counter < gbl_thd_linger)
+        sleep(1);
```

Comment:
Do you mean to increment counter here?

Reply:
maybe just abort if nthrcreates != nretrie to capture cases where server does not wait properly for threads to exit